amgcl/detail/qr.hpp

```diff
@@ -223,10 +223,10 @@ class QR {
             const int row_stride = (Order == row_major ? nmax : 1);
             const int col_stride = (Order == row_major ? 1 : m);
 
-            for(int i = 0, ii = 0; i < n; ++i, ii += row_stride + col_stride)
+            for(int i = 0, ii = 0; i < m; ++i, ii += row_stride + col_stride)
```

Comment:
[zgelsd](http://www.netlib.org/lapack/lapack-3.1.1/html/zgelsd.f.html) routine in lapack seems to do this distinction as well. They factorize `A=QR` or `A=LQ` depending on the shape of the matrix `A`.